### ISSUE TITLE
[Bug?]: Generators not in sync with core features & docs

### ISSUE THREAD

**Issue by mp27** [redwoodjs/redwood#8962]
### What's not working?

Following the tutorial, when it got to the dbAuth part, it didn't work. 
1. The docs for ```dbAuth```  seems to reflect different code that gets generated by the cli for initial ```auth``` function.
2. The web ```auth.js``` file didn't get generated at all while the auth pages tried to import it. I've tried to create the file myself since it's the context, but the request is now asking me for  ```RWJS_API_DBAUTH_URL``` which the docs never mentions.
3. ```App.js``` didn't get updated at all(not sure if it should've been updated with the ```AuthProvider``` in place)
4. In the tutorial, after the setup I don't need to install any packages manually, but there seems to actually be a need for ```@redwoodjs/auth-providers-web``` & ```@redwoodjs/auth```

### How do we reproduce the bug?

Follow the tutorial mentioned in the docs, on the auth implementation it should happen. If this is something that is not related to my environment, it should be easily reproduced.

### What's your environment? (If it applies)

```shell
System:
  OS: Windows, WSL2 - Ubuntu 20
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by jtoar** [comment]
@mp27 I'm assuming you specifically ran `yarn rw setup auth dbAuth` in a new project (v5.4.3), which is at [this point](https://redwoodjs.com/docs/tutorial/chapter4/authentication#auth-setup) in the tutorial, and it didn't work?

1. I'll find time to audit those and report back
2. `web/src/auth.js` is generated:
<img width="2012" alt="image" src="https://github.com/redwoodjs/redwood/assets/32992335/13e95ca6-e2c0-4919-b7da-47c325c5484e">
3.  `App.js` is generated with the `AuthProvider`:
<img width="2012" alt="image" src="https://github.com/redwoodjs/redwood/assets/32992335/f99bc387-3f1c-4d50-9a12-19cabecef38b">
4. I'm not sure what you mean exactly. There is no `@redwoodjs/auth-providers-web` package, only specific auth provider packages, such as `@redwoodjs/auth-dbauth-web`

What version of Redwood are you on? And is what I said accurate to what you did?

**Comment by mp27** [comment]
@jtoar it was 3am for me and thought that maybe I did a mistake, I've did the steps again, I've missed an error last night when running ```yarn rw setup auth dbAuth``` from the [tutorial](https://redwoodjs.com/docs/tutorial/chapter4/authentication#auth-setup). 

Tried with both y/N options, the error is the same. I thought that maybe at some point it's gettings lost because of the changes made to the app and also did the tutorial without making any code changes. 

yarn version: ```3.6.0```

Error:
![image](https://github.com/redwoodjs/redwood/assets/14233831/f3f04e78-54cf-4f53-aead-f177cdd4702b)

The command was ran with dev server on, which also had this error:
![image](https://github.com/redwoodjs/redwood/assets/14233831/0b206772-42b3-4dc4-96d6-60bb8287b4c4)

 BUT, case closed, sorry to waste your time, even tho the dev server was running in node 18...I've use another terminal to run the commands and forgot to switch to node 18, it was still node 16..hence the errors.
